app/controllers/application_controller.rb

```diff
@@ -1,4 +1,22 @@
-# frozen_string_literal: true
-
 class ApplicationController < ActionController::Base
+  def login
+    session[:user_id] = @user.id
+  end
+
+  def logged_in?
+    !!session[:user_id]
+  end
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  def user_name?
+    @user = current_user
+    @user.first_name ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  def set_user
+    @user = User.find_by(id: session[:user_id])
+  end
```

Comment:
Do these functions belong in the application controller? Including the login and logged in functions. Could they possibly be moved to the sessions controller like the logout function was? 
Remember clean code!

Reply:
I looked a but into this, and found that creating a module would help with doing this better, and also would help me learn about modules while doing so.

If you still think it would be better to put the functions in the sessions_controller I can move them, but I feel like this is a good way of doing it

Reply:
Great to hear you're exploring different options. What is the advantage of having these functions within a module over having them in the sessions controller?